docs/DeveloperGuide.adoc

```diff
@@ -358,12 +358,131 @@ image::UndoRedoActivityDiagram.png[width="650"]
 ** Cons: Requires dealing with commands that have already been undone: We must remember to skip these commands. Violates Single Responsibility Principle and Separation of Concerns as `HistoryManager` now needs to do two * different things.
 // end::undoredo[]
 
-// tag::dataencryption[]
-=== [Proposed] Data Encryption
 
-_{Explain here how the data encryption feature will be implemented}_
+// tag::remark[]
+=== Remark feature
+==== Current Implementation
+
+To record more detailed information of a person, a new command `RemarkCommand` is implemented.
+
+The `RemarkCommand` inherits from `UndoableCommands` since it is reasonable to make remark command execute in an _undoable_ way.
+
+`RemarkCommand` which is undoable is implemented this way:
+[source,java]
+----
+public class RemarkCommand extends UndoableCommand {
+    @Override
+    public CommandResult executeUndoableCommand() {
+        // ... remark logic ...
+    }
+}
+----
+
+The model `Person` is modified to have a new field `Remark`.
```

Comment:
Hope to see the Model diagram of this updated. The current model diagram has not been updated to reflect the changes.